### PR TITLE
Bump GCC to 4.9.4

### PR DESCRIPTION
**Pull request by alarcher**
Released Aug 3, 2016, this is the last version in the 4.9 series:

"The GNU Compiler Collection version 4.9.4 has been released.

GCC 4.9.4 is a bug-fix release from the GCC 4.9 branch
containing important fixes for regressions and serious bugs in
GCC 4.9.3 with more than 159 bugs fixed since the previous release.

This is also the last release from the GCC 4.9 branch, GCC continues
to be maintained on the GCC 5 and GCC 6 branches and the development
trunk."

Reference: https://gcc.gnu.org/ml/gcc-announce/2016/msg00002.html

### CODE REVIEW

**Comment by pyhalov** [comment]
Have you checked that you still can build some C++ projects with this gcc, at least clang and boost ?

File: components/developer/gcc49/Makefile [hunk 1]
```diff
@@ -17,8 +17,8 @@
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= gcc
-COMPONENT_VERSION= 4.9.3
-COMPONENT_REVISION=2
+COMPONENT_VERSION= 4.9.4
+COMPONENT_REVISION=0
```
Comment (pyhalov) [inline components/developer/gcc49/Makefile]: remove this line, it's 0 by default

**Comment by pyhalov** [comment]
BTW, developer/clang-34/patches/correct_paths.patch will need fixes, I mean s/4.9.3/4.9.4/g

**Comment by alarcher** [comment]
Some libs are currently being rebuilt on my workstation to test the new version.
I set the label "testing needed" to prevent early merge.

**Comment by pyhalov** [comment]
Please, also bump clang COMPONENT_REVISION (and ensure that it builds :))

**Comment by alarcher** [comment]
I verified that they build but the real question is whether they "run".
Should I expect the gmake check failures in clang-34 ?
Also mpc tests fail due to assertion failure that may be related to precision issues.

**Comment by pyhalov** [comment]
Don't remember exactly, likely, yes

**Comment by alarcher** [comment]
Current status: boost, clang-34 and clang-38 build and passing most tests.
If you can share the results from current build, I can check that no regression exists.
Side notes: 
- it seems there are some threading issues in boost tests.
- clang-38 passes more tests than clang-34.

**Comment by pyhalov** [comment]
Clang 3.4 test results:

---

Testing Time: 203.39s

---

Failing Tests (11):
    Clang :: Sema/wchar.c
    LLVM :: Bindings/Ocaml/executionengine.ml
    LLVM :: Bindings/Ocaml/irreader.ml
    LLVM :: Bindings/Ocaml/target.ml
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyNamedModuleWriteFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyNamedModuleWriteNamedFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyUnnamedModuleWriteNamedFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.ExistingMetadataRetained
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyNamedModuleNoWrite
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyNamedModuleWriteFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyUnnamedModuleWriteToNamedFile

  Expected Passes    : 16015
  Expected Failures  : 84
  Unsupported Tests  : 28
  Unexpected Failures: 11

**Comment by alarcher** [comment]
Clang 3.4.2 test results:

---

Failing Tests (10):
    Clang :: Sema/wchar.c
    LLVM :: Bindings/Ocaml/irreader.ml
    LLVM :: Bindings/Ocaml/target.ml
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyNamedModuleWriteFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyNamedModuleWriteNamedFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.EmptyUnnamedModuleWriteNamedFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.ExistingMetadataRetained
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyNamedModuleNoWrite
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyNamedModuleWriteFile
    LLVM-Unit :: Transforms/DebugIR/Release+Asserts/DebugIRTests/TestDebugIR.NonEmptyUnnamedModuleWriteToNamedFile

  Expected Passes    : 16016
  Expected Failures  : 84
  Unsupported Tests  : 28
  Unexpected Failures: 10

---

Clang 3.8.1 test results:

---

Failing Tests (3):
    LLVM :: CodeGen/AMDGPU/fminnum.ll
    LLVM-Unit :: ADT/Release+Asserts/ADTTests/APIntTest.LargeAPIntConstruction
    LLVM-Unit :: ADT/Release+Asserts/ADTTests/APIntTest.nearestLogBase2

  Expected Passes    : 15344
  Expected Failures  : 147
  Unsupported Tests  : 143
  Unexpected Failures: 3

**Comment by pyhalov** [comment]
LGTM. @xenol: What do you think?

**Comment by alarcher** [comment]
List of tested components:
- boost
- clang-34
- clang-38
- mpich
- openmpi
- valgrind

**Comment by xen0l** [comment]
LGTM

**Pull request merged**